code/modules/mob/living/carbon/xenomorph/Facehuggers.dm

```diff
@@ -219,35 +231,91 @@
 		if(!attached) //Didn't hit anything?
 			i = 5
 			for(M in loc)
-				if(!i) break
+				if(!i)
+					break
 				if(CanHug(M))
 					Attach(M)
 					break
 				i--
 
-/obj/item/clothing/mask/facehugger/proc/Attach(mob/living/M)
-	set waitfor = 0
+/obj/item/clothing/mask/facehugger/throw_at(atom/target, range, speed)
+	. = ..()
+	if(stat == CONSCIOUS)
+		update_stat(UNCONSCIOUS) //Giving it some brief downtime before jumping on someone via movement.
 
-	if(attached || M.status_flags & XENO_HOST || isXeno(M) || loc == M || stat != CONSCIOUS)
+/obj/item/clothing/mask/facehugger/throw_impact(atom/hit_atom, speed)
+	if(stat == DEAD)
 		return
+	if(ismob(hit_atom))
+		if(stat == CONSCIOUS)
+			if(leaping && CanHug(hit_atom)) //Standard leaping behaviour, not attributable to being _thrown_ such as by a Carrier.
+				stat = CONSCIOUS
+				Attach(hit_atom)
+			else if(hit_atom.density) //We hit something, cool.
+				step(src, turn(dir, 180)) //We want the hugger to bounce off if it hits a mob.
+				throwing = FALSE
+				leaping = FALSE
+				addtimer(CALLBACK(src, .proc/fast_activate), 1.5 SECONDS)
 
-	attached++
+	else
+		addtimer(CALLBACK(src, .proc/GoActive), 2 SECONDS)
```

Comment:
This is an issue. It's a very short time to go active, considering the previous behavior was to `GoIdle`, so a random time between 5 and 15 seconds. Try something like 5 or 6 seconds and we'll see how it goes, how does it sound?